File: fontes/frontend/fenix_provider/lib/src/service/compras/compra_tipo_requisicao_service.dart
/*
Title: T2Ti ERP 3.0                                                                
Description: Service relacionado à tabela [COMPRA_TIPO_REQUISICAO] 
                                                                                
The MIT License                                                                 
                                                                                
Copyright: Copyright (C) 2021 T2Ti.COM                                          
                                                                                
Permission is hereby granted, free of charge, to any person                     
obtaining a copy of this software and associated documentation                  
files (the "Software"), to deal in the Software without                         
restriction, including without limitation the rights to use,                    
copy, modify, merge, publish, distribute, sublicense, and/or sell               
copies of the Software, and to permit persons to whom the                       
Software is furnished to do so, subject to the following                        
conditions:                                                                     
                                                                                
The above copyright notice and this permission notice shall be                  
included in all copies or substantial portions of the Software.                 
                                                                                
THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,                 
EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES                 
OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND                        
NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT                     
HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,                    
WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING                    
FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR                   
OTHER DEALINGS IN THE SOFTWARE.                                                 
                                                                                
       The author may be contacted at:                                          
           t2ti.com@gmail.com                                                   
                                                                                
@author Albert Eije (alberteije@gmail.com)                    
@version 1.0.0
*******************************************************************************/
import 'dart:convert';
import 'dart:typed_data';

import 'package:http/http.dart' show Client;

import 'package:fenix/src/service/service_base.dart';
import 'package:fenix/src/model/filtro.dart';
import 'package:fenix/src/model/model.dart';

/// classe responsável por requisições ao servidor REST
class CompraTipoRequisicaoService extends ServiceBase {
  var clienteHTTP = Client();

  Future<List<CompraTipoRequisicao>?> consultarLista({Filtro? filtro}) async {
    List<CompraTipoRequisicao> listaCompraTipoRequisicao = [];

    try {
      tratarFiltro(filtro, '/compra-tipo-requisicao/');
	  	
      final response = await clienteHTTP.get(Uri.tryParse(url)!, headers: ServiceBase.cabecalhoRequisicao);

      if (response.statusCode == 200) {
        if (response.headers["content-type"]!.contains("html")) {
          tratarRetornoErro(response.body, response.headers);
          return null;
        } else {
          var parsed = json.decode(response.body) as List<dynamic>;
          for (var compraTipoRequisicao in parsed) {
            listaCompraTipoRequisicao.add(CompraTipoRequisicao.fromJson(compraTipoRequisicao));
          }
          return listaCompraTipoRequisicao;
        }
      } else {
        tratarRetornoErro(response.body, response.headers);
        return null;
      }
    } on Exception catch (e) {
      tratarRetornoErro(null, null, exception: e);
    }
    return null;
  }

  Future<CompraTipoRequisicao?> consultarObjeto(int id) async {
    try {
      final response = await clienteHTTP.get(Uri.tryParse('$endpoint/compra-tipo-requisicao/$id')!, headers: ServiceBase.cabecalhoRequisicao);

      if (response.statusCode == 200) {
        if (response.headers["content-type"]!.contains("html")) {
          tratarRetornoErro(response.body, response.headers);
          return null;
        } else {
          var compraTipoRequisicaoJson = json.decode(response.body);
          return CompraTipoRequisicao.fromJson(compraTipoRequisicaoJson);
        }
      } else {
        tratarRetornoErro(response.body, response.headers);
        return null;
      }
    } on Exception catch (e) {
      tratarRetornoErro(null, null, exception: e);
    }
    return null;
  }

  Future<CompraTipoRequisicao?> inserir(CompraTipoRequisicao compraTipoRequisicao) async {
    try {
      final response = await clienteHTTP.post(
        Uri.tryParse('$endpoint/compra-tipo-requisicao')!,
        headers: ServiceBase.cabecalhoRequisicao,
        body: compraTipoRequisicao.objetoEncodeJson(compraTipoRequisicao),
      );

      if (response.statusCode == 200 || response.statusCode == 201) {
        if (response.headers["content-type"]!.contains("html")) {
          tratarRetornoErro(response.body, response.headers);
          return null;
        } else {
          var compraTipoRequisicaoJson = json.decode(response.body);
          return CompraTipoRequisicao.fromJson(compraTipoRequisicaoJson);
        }
      } else {
        tratarRetornoErro(response.body, response.headers);
        return null;
      }
    } on Exception catch (e) {
      tratarRetornoErro(null, null, exception: e);
    }
    return null;
  }

  Future<CompraTipoRequisicao?> alterar(CompraTipoRequisicao compraTipoRequisicao) async {
    try {
      var id = compraTipoRequisicao.id;
      final response = await clienteHTTP.put(
        Uri.tryParse('$endpoint/compra-tipo-requisicao/$id')!,
        headers: ServiceBase.cabecalhoRequisicao,
        body: compraTipoRequisicao.objetoEncodeJson(compraTipoRequisicao),
      );

      if (response.statusCode == 200) {
        if (response.headers["content-type"]!.contains("html")) {
          tratarRetornoErro(response.body, response.headers);
          return null;
        } else {
          var compraTipoRequisicaoJson = json.decode(response.body);
          return CompraTipoRequisicao.fromJson(compraTipoRequisicaoJson);
        }
      } else {
        tratarRetornoErro(response.body, response.headers);
        return null;
      }
    } on Exception catch (e) {
      tratarRetornoErro(null, null, exception: e);
    }
    return null;
  }

  Future<bool?> excluir(int id) async {
    try {
      final response = await clienteHTTP.delete(
        Uri.tryParse('$endpoint/compra-tipo-requisicao/$id')!,
        headers: ServiceBase.cabecalhoRequisicao,
      );

      if (response.statusCode == 200) {
        return true;
      } else {
        tratarRetornoErro(response.body, response.headers);
        return null;
      }
    } on Exception catch (e) {
      tratarRetornoErro(null, null, exception: e);
    }
    return null;
  }
  
  Future<Uint8List?> visualizarPdf({Filtro? filtro}) async {
    try {
      tratarFiltroRelatorio(filtro, 'CompraTipoRequisicao');
      final response = await clienteHTTP.get(Uri.tryParse(urlRelatorios)!);
      if (response.statusCode == 200) {
        if (response.headers["content-type"]!.contains("html")) {
          tratarRetornoErro(response.body, response.headers);
          return null;
        } else {
          final pdf = response.bodyBytes;
          return pdf;        
        }
      } else {
        tratarRetornoErro(response.body, response.headers);
        return null;
      }
    } on Exception catch (e) {
      tratarRetornoErro(null, null, exception: e);
    }
    return null;
  }

  Future<void> visualizarPdfWeb({Filtro? filtro}) async {
    try {
      tratarFiltroRelatorio(filtro, 'CompraTipoRequisicao');
      visualizarRelatorioPdfWeb(filtro, 'Relatório Compra Tipo Requisicao');
    } on Exception catch (e) {
      tratarRetornoErro(null, null, exception: e);
    }
  }
  
}

File: fontes/backend/c-sharp/T2TiERPFenix-NHibernate/T2TiERPFenix/Services/Compras/CompraTipoRequisicaoService.cs
/*******************************************************************************
Title: T2Ti ERP Fenix                                                                
Description: Service relacionado à tabela [COMPRA_TIPO_REQUISICAO] 
                                                                                
The MIT License                                                                 
                                                                                
Copyright: Copyright (C) 2020 T2Ti.COM                                          
                                                                                
Permission is hereby granted, free of charge, to any person                     
obtaining a copy of this software and associated documentation                  
files (the "Software"), to deal in the Software without                         
restriction, including without limitation the rights to use,                    
copy, modify, merge, publish, distribute, sublicense, and/or sell               
copies of the Software, and to permit persons to whom the                       
Software is furnished to do so, subject to the following                        
conditions:                                                                     
                                                                                
The above copyright notice and this permission notice shall be                  
included in all copies or substantial portions of the Software.                 
                                                                                
THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,                 
EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES                 
OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND                        
NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT                     
HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,                    
WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING                    
FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR                   
OTHER DEALINGS IN THE SOFTWARE.                                                 
                                                                                
       The author may be contacted at:                                          
           t2ti.com@gmail.com                                                   
                                                                                
@author Albert Eije (alberteije@gmail.com)                    
@version 1.0.0
*******************************************************************************/
using NHibernate;
using System.Collections.Generic;
using T2TiERPFenix.Models;
using T2TiERPFenix.NHibernate;

namespace T2TiERPFenix.Services
{
    public class CompraTipoRequisicaoService
    {

        public IEnumerable<CompraTipoRequisicao> ConsultarLista()
        {
            IList<CompraTipoRequisicao> Resultado = null;
            using (ISession Session = NHibernateHelper.GetSessionFactory().OpenSession())
            {
                NHibernateDAL<CompraTipoRequisicao> DAL = new NHibernateDAL<CompraTipoRequisicao>(Session);
                Resultado = DAL.Select(new CompraTipoRequisicao());
            }
            return Resultado;
        }

        public IEnumerable<CompraTipoRequisicao> ConsultarListaFiltro(Filtro filtro)
        {
            IList<CompraTipoRequisicao> Resultado = null;
            using (ISession Session = NHibernateHelper.GetSessionFactory().OpenSession())
            {
                var consultaSql = "from CompraTipoRequisicao where " + filtro.Where;
                NHibernateDAL<CompraTipoRequisicao> DAL = new NHibernateDAL<CompraTipoRequisicao>(Session);
                Resultado = DAL.SelectListaSql<CompraTipoRequisicao>(consultaSql);
            }
            return Resultado;
        }
		
        public CompraTipoRequisicao ConsultarObjeto(int id)
        {
            CompraTipoRequisicao Resultado = null;
            using (ISession Session = NHibernateHelper.GetSessionFactory().OpenSession())
            {
                NHibernateDAL<CompraTipoRequisicao> DAL = new NHibernateDAL<CompraTipoRequisicao>(Session);
                Resultado = DAL.SelectId<CompraTipoRequisicao>(id);
            }
            return Resultado;
        }
		
        public void Inserir(CompraTipoRequisicao objeto)
        {
            using (ISession Session = NHibernateHelper.GetSessionFactory().OpenSession())
            {
                NHibernateDAL<CompraTipoRequisicao> DAL = new NHibernateDAL<CompraTipoRequisicao>(Session);
                DAL.SaveOrUpdate(objeto);
                Session.Flush();
            }
        }

        public void Alterar(CompraTipoRequisicao objeto)
        {
            using (ISession Session = NHibernateHelper.GetSessionFactory().OpenSession())
            {
                NHibernateDAL<CompraTipoRequisicao> DAL = new NHibernateDAL<CompraTipoRequisicao>(Session);
                DAL.SaveOrUpdate(objeto);
                Session.Flush();
            }
        }

        public void Excluir(CompraTipoRequisicao objeto)
        {
            using (ISession Session = NHibernateHelper.GetSessionFactory().OpenSession())
            {
                NHibernateDAL<CompraTipoRequisicao> DAL = new NHibernateDAL<CompraTipoRequisicao>(Session);
                DAL.Delete(objeto);
                Session.Flush();
            }
        }
		
    }

}